From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Dominik Csapak <d.csapak@proxmox.com>
Date: Thu, 11 May 2017 10:34:10 +0200
Subject: [PATCH] change scaling when toggling fullscreen

when activating fullscreen, we change the scaling to 'scale',
and changing it back when leaving fullscreen

Signed-off-by: Dominik Csapak <d.csapak@proxmox.com>
---
 app/ui.js | 11 +++++++++++
 1 file changed, 11 insertions(+)

diff --git a/app/ui.js b/app/ui.js
index 11fad2d..b40d1f8 100644
--- a/app/ui.js
+++ b/app/ui.js
@@ -1240,6 +1240,13 @@ const UI = {
             } else if (document.msExitFullscreen) {
                 document.msExitFullscreen();
             }
+
+            // when changing from fullscreen to window,
+            // re enable auto resize
+            if (WebUtil.getQueryVar('resize') !== 'scale') {
+                UI.forceSetting('resize', 'off');
+                UI.enableSetting('resize');
+            }
         } else {
             if (document.documentElement.requestFullscreen) {
                 document.documentElement.requestFullscreen();
@@ -1250,7 +1257,11 @@ const UI = {
             } else if (document.body.msRequestFullscreen) {
                 document.body.msRequestFullscreen();
             }
+            // we want scaling in fullscreen mode
+            UI.forceSetting('resize', 'scale');
+            UI.enableSetting('resize');
         }
+        UI.applyResizeMode();
         UI.updateFullscreenButton();
     },
 
